pkg/fixtures/vmindexreport/generator.go

```diff
@@ -0,0 +1,241 @@
+// Package vmindexreport provides utilities for generating fake VM index reports
+// for testing and load testing purposes.
+package vmindexreport
+
+import (
+	"fmt"
+	"math/rand"
+	"regexp"
+	"strconv"
+	"strings"
+
+	v4 "github.com/stackrox/rox/generated/internalapi/scanner/v4"
+	v1 "github.com/stackrox/rox/generated/internalapi/virtualmachine/v1"
+)
+
+const (
+	// MockDigest is kept in sync with pkg/virtualmachine/enricher/enricher_impl.go
+	MockDigest = "sha256:900dc0ffee900dc0ffee900dc0ffee900dc0ffee900dc0ffee900dc0ffee900d"
+
+	// MockDigestWithRegistry is the digest with a mock registry prefix for use with name.NewDigest()
+	MockDigestWithRegistry = "vm-registry/repository@sha256:900dc0ffee900dc0ffee900dc0ffee900dc0ffee900dc0ffee900dc0ffee900d"
+)
+
+// numericRegex matches sequences of digits in a version string.
+var numericRegex = regexp.MustCompile(`\d+`)
+
+// NormalizeRPMVersion parses an RPM version string and returns a 10-element int32 slice.
+// Only the first 3 numeric components (major, minor, patch) are extracted since those
+// are what matter for vulnerability matching. The rest are zeros.
+// The epoch prefix (e.g., "1:") is stripped before parsing.
+// Example: "1:1.54.0-3.el9_7" → [1, 54, 0, 0, 0, 0, 0, 0, 0, 0]
+func NormalizeRPMVersion(version string) []int32 {
+	result := make([]int32, 10)
+
+	// Strip epoch prefix if present (e.g., "1:" in "1:1.54.0-3.el9_7")
+	if idx := strings.Index(version, ":"); idx != -1 {
+		version = version[idx+1:]
+	}
+
+	// Extract first 3 numeric sequences from the version string
+	matches := numericRegex.FindAllString(version, 3)
```

Comment:
Some versions don't have a patch element, see this example from a real index report:

```json
{
        "id": 131,
        "name": "pigz",
        "version": "2.8-1.el9",
        ...
}
```

That would make this function return misleading data, but perhaps this is not a problem?

Reply:
Thanks for the explanation. I would still add a comment that we know this is flawed but it's fine. But I leave that to your preference!

Reply:
Thanks! I can add a comment if there is a rebase or another commit needed. If CI is happy with the current state, I would prefer merging as is.